add project to GH

Graev opened the pull request:
Развернут сервер на Express
команда npm run start запускает сервер на localhost:3000;
команда npm run dev запускает сервер на localhost:3000 с хот релоудом;
адрес localhost:3000 загружает фронтенд проекта Mesto, который вы делали;
в ответ на запрос GET localhost:3000/users сервер вернёт JSON-объект из файла users.json;
в ответ на запрос GET localhost:3000/cards сервер вернёт JSON-объект из файла cards.json;
в ответ на запрос GET localhost:3000/users/8340d0ec33270a25f2413b69, сервер вернёт JSON-объект пользователя с переданным после /users идентификатором;
если пользователя с запрошенным идентификатором нет, API должен возвращать 404 статус ответа и JSON: { "message": "Нет пользователя с таким id" };
при запросе на несуществующий адрес, API должен возвращать 404 статус ответа и JSON: { "message": "Запрашиваемый ресурс не найден" }.

Graev committed:
add project to GH

koras commented on routes/cards.js:
```diff
@@ -0,0 +1,15 @@
+const fs = require('fs');
+
+routerCards.get('/', (req, res) => {
+  fs.readFile(path.join(__dirname, '..', 'data', 'cards.json'), { encoding: 'utf8' }, (err, data) => {
```
Зачем вы подключаете как файл. У вас же есть require для этого. 
Сейчас вместо карточек возвращается каша непонятная.

koras commented on routes/users.js:
```diff
@@ -0,0 +1,36 @@
+const fs = require('fs');
+
+routerUsers.get('/', (req, res) => {
+  fs.readFile(path.join(__dirname, '..', 'data', 'users.json'), { encoding: 'utf8' }, (err, data) => {
```
Тоже самое и здесь.
Это не правильное решение

koras commented on routes/users.js:
```diff
@@ -0,0 +1,36 @@
+    }
+    const user = JSON.parse(data).filter((i) => {
+      if (i._id == req.params.id) {
+        return true;
```
достаточно просто вернуть объект с пользователем

koras reviewed (changes requested):
Надо довести работу до ума.

Graev committed:
Error correction

Справил метод подключение JSON файлов.
Изменил метод поиска пользователя по id.

koras commented on routes/users.js:
```diff
@@ -0,0 +1,19 @@
+const routerUsers = require('express').Router();
+const path = require('path');
+const userJSON = require(path.join(__dirname, '..', 'data', 'users.json'));
```
Зачем __dirname, опять вы велосипед изобретаете

koras reviewed (approved):
Работа принимается. Все ошибки исправлены

Graev committed:
sprint #13

в .eslintrc добавлено исключение для _id;
Node.js приложение подключается к серверу Mongo по адресу mongodb://localhost:27017/mestodb;
создана схема и модель пользователя с полями name, about и avatar, поля корректно валидируются;
создана схема и модель карточки с полями name, link, owner, likes и createdAt, поля корректно валидируются;
запрос на GET /users возвращает всех пользователей из базы;
запрос GET /users/:userId возвращает конкретного пользователя;
запрос POST /users создаёт пользователя;
запрос GET /cards возвращает все карточки всех пользователей;
запрос POST /cards создаёт карточку;
запрос PATCH /users/me обновляет профиль;
запрос PATCH /users/me/avatar обновляет аватар;
запрос PUT /cards/:cardId/likes — поставит лайк карточке;
запрос DELETE /cards/:cardId/likes — уберет лайк карточке;
если в любом из перечисленных запросов что-то идёт не так, сервер возвращает ошибку, не забывайте добавлять статус;
папки public и data удалены;
